ui/StatusQ/src/StatusQ/Components/StatusMessageHeader.qml

```diff
@@ -61,81 +59,147 @@ Item {
                     root.clicked(this, mouse)
                 }
             }
-        }       
-        StatusBaseText {
-            id: messageOriginInfo
-            verticalAlignment: Text.AlignVCenter
-            visible: text
-            color: Theme.palette.baseColor1
-            font.pixelSize: Theme.asideTextFontSize
-            text: root.messageOriginInfo
         }
-        StatusContactVerificationIcons {
-            id: verificationIcons
-            visible: !root.amISender
-            isContact: root.isContact
-            trustIndicator: root.trustIndicator
+
+        Loader {
+            id: messageOriginInfoLoader
+            active: root.messageOriginInfo
+            asynchronous: true
+            sourceComponent: Component {
```

Comment:
Small tip: `Component {` is not needed. It will remove one level of indentation.

Reply:
done, thanks!